Simulator/Recorders/XmlRecorder.cpp

```diff
@@ -36,6 +36,14 @@ XmlRecorder::XmlRecorder() :
 void XmlRecorder::init()
 {
    resultOut_.open(resultFileName_.c_str());
+   // check the file extension is .xml
+   string suffix = ".xml";
+   if ((resultFileName_.size() <= suffix.size())
+       || (resultFileName_.compare(resultFileName_.size() - suffix.size(), suffix.size(), suffix)
+           != 0)) {
+      perror("the file extention is not .xml ");
+      exit(EXIT_FAILURE);
+   }
```

Comment:
Move this to before opening the file.

Reply:
Yes. It makes more sense to move it before testing if the file can be opened.